waterspout_api/views.py

```diff
@@ -35,6 +40,96 @@ def get_user_info_dict(user, token):
 		}
 
 
+class GetPasswordReset(APIView):
+    """
+    Returns a link to reset password
+    """
+    serializer_class = serializers.EmailSerializer
+
+    # this allows for users to not be signed in
+    permission_classes = [AllowAny]
+
+    def post(self, request):
+
+        serializers = self.serializer_class(data=request.data)
+        serializers.is_valid(raise_exception=True)
+
+        user_email = serializers.data["email"]
+        user = get_user_model().objects.filter(email=user_email).first()
+
+        # if the email is found
+        if user:
+            encoded_pk = urlsafe_base64_encode(force_bytes(user.pk))
+            new_token = PasswordResetTokenGenerator().make_token(user)
+
+            domain = request.get_host()
+            # temporary while I do testing
+            reset_url = f'''
+					<p>Click the following button to reset your password:</p>
+					<a href="http://localhost:5173/#/password-reset?encoded_pk={encoded_pk}&token={new_token}" style="text-decoration: none;">
```

Comment:
That URL of localhost:5173 is going to fail in production - it should ideally be a variable from local settings, but I don't think we actually limit to just one URL (we may want to). For now, I'd suggest hardcoding it to:

`https://openag.ucmerced.edu/api/password-reset?`  and then keeping the URL parameters as-is.